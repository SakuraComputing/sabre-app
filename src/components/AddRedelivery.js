import React from 'react';
import { connect } from 'react-redux';
import RedeliveryForm from './RedeliveryForm';
import { createRedelivery } from "../actions/redelivery";
import UrlConstants from '../../config/url-constants';

export class AddRedelivery extends React.Component {
    onSubmit = (redelivery) => {
      this.props.createRedelivery(redelivery);
      this.props.history.push(UrlConstants.REDELIVERY_HOME);
    };
    render() {
        return (
            <div>
                <h2>Add Redelivery</h2>
                <RedeliveryForm
                    onSubmit = {this.onSubmit}
                />
            </div>
        );
    }
}

export const mapDispatchToProps = (dispatch) => ({
    createRedelivery: (redelivery) => dispatch(createRedelivery(redelivery))
});
export default connect(undefined, mapDispatchToProps)(AddRedelivery);