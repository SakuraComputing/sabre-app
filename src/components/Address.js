import React from 'react';

const Address = ( { address } ) => (
    <table className='address-panel'>
        <tbody>
            <tr>{address.addressName}</tr>
            <tr>{address.addressLine1}</tr>
            <tr>{address.addressLine2}</tr>
            <tr>{address.town}</tr>
            <tr>{address.county}</tr>
            <tr>{address.telephone}</tr>
        </tbody>
    </table>
);
export default Address;
