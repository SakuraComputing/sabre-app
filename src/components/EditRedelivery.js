import React from 'react';
import { connect } from 'react-redux';
import { updateRedelivery, deleteRedelivery} from '../actions/redelivery';
import RedeliveryForm from './RedeliveryForm';
import UrlConstants from "../../config/url-constants";

const REDELIVERY_URL = UrlConstants.REDELIVERY_ID;

export class EditRedelivery extends React.Component {

    onSubmit = (redelivery) => {
        this.props.updateRedelivery(this.props.redelivery.id, redelivery);
        this.props.history.push('/redelivery');
    };

    onDelete = (id) => {
        this.props.deleteRedelivery( this.props.cardId);
        this.props.history.push('/redelivery');
    };

    render() {
        return (
            <div>
                <h2>Edit Redelivery</h2>
                <RedeliveryForm
                    redelivery={this.props.redelivery}
                    onSubmit={this.onSubmit}
                />
                <button
                    onClick={this.onDelete}>
                    Delete Redelivery
                </button>
            </div>
        );
    }
}

export const mapStateToProps = ( state, props ) => ({
   redelivery: state.redeliveries.find((redelivery) => {
       return redelivery.cardId === props.match.params.id;
   })
});

export const mapDispatchToProps = (dispatch) => ({
    updateRedelivery: (id, redelivery) => dispatch(updateRedelivery(id, redelivery)),
    deleteRedelivery: (data) => dispatch(deleteRedelivery(data))
});
export default connect(mapStateToProps, mapDispatchToProps)(EditRedelivery);
