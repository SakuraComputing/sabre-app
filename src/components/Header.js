import React from 'react';
import { NavLink } from 'react-router-dom';
import { Navbar } from 'react-bootstrap';

const Header = () => (
    <div className='menu-container'>
        <Navbar className="navbar navbar-default navbar-static-top" >
            <ul>

                <li>
                    <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
                </li>
                <li>
                    <NavLink to="/tracking" activeClassName="is-active">Parcel Tracking</NavLink>
                </li>
                <li>
                    <NavLink to="/redelivery" activeClassName="is-active">Redelivery</NavLink>
                </li>
            </ul>
        </Navbar>
        <h1>Yodel Web Portal</h1>
    </div>
);
export default Header;
