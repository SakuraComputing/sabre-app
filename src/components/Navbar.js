import React from 'react';

export default class Navbar extends React.Component {

    constructor(props) {
        super(props);
    }

    onHamburgerClick = () =>  {
      this.props.onHamburgerClick && this.props.onHamburgerClick();
    };

    render(){
        return (
            <nav className="navbar navbar-default navbar-fixed-top">
                <div className="container-fluid">
                    <div className="navbar-header">
                        <button type="button" className="navbar-btn navbar-toggle collapsed" onClick={this.onHamburgerClick}>
                            <span className="sr-only">Toggle Navigation</span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                            <span className="icon-bar"></span>
                        </button>
                    </div>
                    {this.props.children}
                </div>
            </nav>
        );
    }
}