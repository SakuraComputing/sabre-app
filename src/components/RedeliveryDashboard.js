import React from 'react';
import RedeliveryHeader from './RedeliveryHeader';
import RedeliveryList from './RedeliveryList';
import { NavLink } from 'react-router-dom';

const RedeliveryDashboard = () =>(
    <header>
        <h2>Redeliveries</h2>
        <p></p>
        <NavLink to="/create" activeClassName="is-active">Add Redelivery</NavLink>
        <div>
            <RedeliveryHeader/>
            <RedeliveryList />
        </div>

    </header>
);
export default RedeliveryDashboard;


