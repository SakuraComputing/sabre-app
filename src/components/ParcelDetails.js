import React from 'react';
import Address from "./Address";

const ParcelDetails = ({ parcel, address }) => (
    <table className='parcel-details' frameBorder="1">
        <tbody>
            <tr>Address:</tr>
            <tr className="address-details" >{address ? <Address address={address}/> : 'Address not found'}</tr>
            <tr>Service Code: {parcel.serviceCode}</tr>
        </tbody>
    </table>
);
export default ParcelDetails;
