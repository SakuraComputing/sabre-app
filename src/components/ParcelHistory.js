import React from 'react';
import parcelData from '../../data/parcel.json';
import trackData from '../../data/tracks.json';
import addressData from '../../data/address.json';
import ParcelDetails from "./ParcelDetails";
import TrackingHistory from "./TrackingHistory";

class ParcelHistory extends React.Component {

    constructor(props) {
        super(props);
        this.state = {parcelId: '', error: '', selectedParcel: '', selectedTracks: '', selectedAddress: ''};
    }

    onSubmit = (e) => {
        e.preventDefault();

        let parcelId = this.state.parcelId;
        if (parcelId) {
            if(this.onValidate()) {
                const selectedParcel = parcelData["parcels"].find(parcel => parcel.parcelId === parcelId);
                if (selectedParcel) {
                    let addressNo = selectedParcel.addressNo;
                    var selectedAddress = addressData["addresses"].find(address => address.addressNo === addressNo);
                    var selectedTracks = trackData["tracks"].filter(track => track.parcelId === parcelId);
                }
                let error = selectedParcel ? null : 'parcel not found';

                this.setState(() => ({
                    selectedParcel: selectedParcel,
                    error: error,
                    selectedTracks: selectedTracks,
                    selectedAddress: selectedAddress
                }));
            }
        } else {
            this.setState(() => ({ error: 'Please provide a parcel Id' }));
        }
    };

    onParcelChange = (e) => {
        const parcelId = e.target.value;
        this.setState(() => ({ parcelId }));
    };

    onClear = () => {
        this.setState(()=> ({ parcelId: '', error: null, selectedParcel: '', selectedTracks: '', selectedAddress: '' }))
    };

    onValidate = (e) => {
        let parcelId = this.state.parcelId;
        let formIsValid = true;

        if(parcelId.length < 12 || parcelId.length > 35) {
            this.setState(() => ({ error: 'Parcel must be between 12 and 35 chars' }));
            formIsValid = false
        }
        return formIsValid;
    }
    ;

    render() {
        let id = 'parcelId';
        return (
            <div>
                <form className="quick-form" onSubmit={this.onSubmit} >
                    <h2>Parcel History</h2>
                    <div className="search-bar">
                        Parcel Id:
                        <input
                            onChange={this.onParcelChange}
                            name={id}
                            id={id}
                            value={this.state.parcelId}
                            placeholder="Parcel Id"
                        />
                        <button className="submit-button">
                            Search
                        </button>
                        <button type="reset" className="clear-button" onClick={this.onClear}>
                            Clear Search
                        </button>
                    </div>
                    {this.state.selectedParcel ? <ParcelDetails parcel={this.state.selectedParcel} address={this.state.selectedAddress}/> : ''}
                    {this.state.selectedTracks ? <TrackingHistory tracks={this.state.selectedTracks} /> : ''}
                </form>
                {this.state.error && <div className="error-message">{this.state.error}</div>}
            </div>
        )
    };
};
export default ParcelHistory;

