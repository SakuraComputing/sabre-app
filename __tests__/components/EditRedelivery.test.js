import React from 'react';
import { shallow } from 'enzyme';
import { EditRedelivery } from '../../src/components/EditRedelivery';
import redeliveries from '../fixtures/redeliveries';
import {mapDispatchToProps} from "../../src/components/EditRedelivery";

describe('edit redelivery page', () => {

    let wrapper, updateRedelivery, deleteRedelivery, history;

    beforeEach(() => {
        updateRedelivery = jest.fn();
        deleteRedelivery = jest.fn();
        history = { push: jest.fn() };
        wrapper = shallow(<EditRedelivery
            updateRedelivery={updateRedelivery}
            deleteRedelivery={deleteRedelivery}
            history={history}
            redelivery={redeliveries[2]}
        />);
    });

    it('should render the form correctly', () => {
      expect(wrapper).toMatchSnapshot();
   });
   it('should handle update redelivery', () => {
       wrapper.find('RedeliveryForm').prop('onSubmit')(redeliveries[2]);
       // expect(history.push).toHaveBeenCalledWith('/redelivery');
       // expect(editRedelivery).toHaveBeenCalledWith(redeliveries[2].id,redeliveries[2])
   });
    it('should handle removeRedelivery', () => {
        // wrapper.find('button').simulate('click');
        // expect(history.push).toHaveBeenLastCalledWith('/redelivery');
        // expect(removeRedelivery).toHaveBeenLastCalledWith({
        //     id: redeliveries[2].id
        // });
    });

    describe('maps dispatch to props',() => {
        const id = 1;

        it('should dispatch redelivery to props', () => {
            const redelivery = {
                id: 1,
                addressNo: 0,
                cardId: '99999',
                parcelId: 'JD000',
                latestScan: 0,
                requestDate: 0
            };
            const mockDispatchEdit = jest.fn();
            const actionProps = mapDispatchToProps(mockDispatchEdit);

            actionProps.updateRedelivery(id, redelivery);
            expect(mockDispatchEdit).toHaveBeenCalledTimes(1);
        });
        it('should dispatch remove redelivery',() => {
            const mockDispatchRemove = jest.fn();
            const actionProps = mapDispatchToProps(mockDispatchRemove);
            actionProps.deleteRedelivery(id);
            expect(mockDispatchRemove).toHaveBeenCalledTimes(1);
        });
    });
});
