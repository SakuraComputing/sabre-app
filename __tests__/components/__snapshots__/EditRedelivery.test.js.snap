// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit redelivery page should render the form correctly 1`] = `
<div>
  <h2>
    Edit Redelivery
  </h2>
  <RedeliveryForm
    onSubmit={[Function]}
    redelivery={
      Object {
        "addressNo": 7,
        "cardId": "376363",
        "id": "3",
        "latestScan": 1325376000000,
        "parcelId": "JD0002289756470005",
        "requestDate": 1325376000000,
      }
    }
  />
  <button
    onClick={[Function]}
  >
    Delete Redelivery
  </button>
</div>
`;
